calculate distance in walking time (I have the code to do it, but I need to put it into a function that can be used in to_df)

acoh64 opened the pull request.

acoh64 committed:
calculate distance in walking time

anandijain commented:
nice. yeah move it to it's own function then we need to add a test. 

then you call that function on each of the lat lons returned by overpass and add this as a new column to the view_df

anandijain commented on test/basic_overpass_query.jl:
```diff
@@ -29,14 +31,20 @@ out;
+route_url = "http://router.project-osrm.org/table/v1/foot/-71.105650,42.363740;-71.106120,42.358030"
+response = HTTP.post(route_url)
+j = JSON3.read(response.body)
+j.durations[1][2] #this is the time (in seconds) it takes to go from the first lat/long pair to the second lat/long pair
```
since we have a hardcoded query url, we should expect this to be relatively stable.

I would add something like 
```julia
@test j.durations[1][2] == 50
```
or if that ends up failing stochastically 
```julia
@test isapprox(j.durations[1][2], 50;atol=something)
```
testing that the json returned 50 ± something seconds

acoh64 committed:
added function and changed to sort by new distance

acoh64 commented:
I added a function `walking_time_distance` and added a new column to the df returned by df called `time_distance`. I also changed the `overpass_api` to have it sort by this new distance instead of the haversine distance

acoh64 commented:
@anandijain since the main change is to the `to_df` function do I need to add a new test (`to_df ` is already tested)? not really sure the proper way to write tests

anandijain commented on test/overpass_api.jl:
```diff
@@ -13,8 +13,8 @@ example_query = "https://www.google.com/maps/dir/?api=1&origin=<lat, long>&desti
-    view_df = df[:, [:id, :distance, :lat, :lon, :maps_url]]
-    sort!(view_df, :distance)
+    view_df = df[:, [:id, :distance, :time_distance, :lat, :lon, :maps_url]]
+    sort!(view_df, :time_distance)
```
dope!

anandijain committed:
Merge branch 'main' into ac/distances

anandijain commented:
yeah just test the new function for now. we can resolve #5 (testing `to_df`) later. you'll see that since I merged my location from device pr, there are merge conflicts. i just pushed a commit that resolves those. 

i like that you only make one request to OSRM instead of making one for each row of the df, thats really nice

anandijain commented:
also note that `test/basic_overpass_query.jl` does not get `include`d in the `test/runtests.jl` file (meaning that nothing there actually gets tested by CI)

acoh64 commented:
Yeah at first I was making one request each but it was super slow so I changed it to one request in total. Much faster and get same results.

anandijain commented:
im going to quickly try the api locally, then we can merge. i think it would be good for you to get good at CI and testing of julia packages (test driven development) so maybe we can assign #5 to you. 

https://github.com/ndortega/Oxygen.jl/blob/master/test/runtests.jl#L311 here and below are examples of how Oxygen.jl tests itself. happy to help you thru it

acoh64 commented:
Sounds good. I'm going to sleep now but we can go through it tomorrow night when we pee.

anandijain commented:
<img width="551" alt="image" src="https://user-images.githubusercontent.com/33790004/220539314-2161bfb6-c4c5-4a44-9900-e4b7e57318ab.png">

incredible! 
sorting by time definitely seems to improve the sanity of the results a good bit! of course we still havent implemented logic about checking whether these places are actually open or not. but this is definitely good to merge

anandijain commented:
oh i forgot about the tests, just add those when you do #5 i guess
